diff --git a/templates/nsis/installSection.nsh b/templates/nsis/installSection.nsh
index 4508b2c..0463c18 100644
--- a/templates/nsis/installSection.nsh
+++ b/templates/nsis/installSection.nsh
@@ -9,5 +9,8 @@ StrCpy $appExe "$INSTDIR\${APP_EXECUTABLE_FILENAME}"
 # must be called before uninstallOldVersion
 !insertmacro setLinkVars
-
+!ifmacrodef customInstallBeforeCheckAppRunning
+  !insertmacro customInstallBeforeCheckAppRunning
+!endif
+
 !ifdef ONE_CLICK
   !ifdef HEADER_ICO
@@ -38,3 +41,6 @@ ${ifNot} ${UAC_IsInnerInstance}
 !endif
-
+!ifmacrodef customInstallBeforeInstallationChanges
+  !insertmacro customInstallBeforeInstallationChanges
+!endif
+
 Var /GLOBAL keepShortcuts
@@ -52,5 +58,8 @@ ${if} $isTryToKeepShortcuts == "true"
   ${endIf}
 ${endif}
-
+!ifmacrodef customInstallBeforeRemovingPreviousVersion
+  !insertmacro customInstallBeforeRemovingPreviousVersion
+!endif
+
 !insertmacro uninstallOldVersion SHELL_CONTEXT
 !insertmacro handleUninstallResult SHELL_CONTEXT
@@ -65,5 +74,8 @@ SetOutPath $INSTDIR
   File /oname=uninstallerIcon.ico "${UNINSTALLER_ICON}"
 !endif
-
+!ifmacrodef customInstallBeforeApplicationFiles
+  !insertmacro customInstallBeforeApplicationFiles
+!endif
+
 !insertmacro installApplicationFiles
 !insertmacro registryAddInstallInfo
diff --git a/templates/nsis/assistedInstaller.nsh b/templates/nsis/assistedInstaller.nsh
index 3c6bb19..91f87e8 100644
--- a/templates/nsis/assistedInstaller.nsh
+++ b/templates/nsis/assistedInstaller.nsh
@@ -12,4 +12,8 @@
   !ifmacrodef licensePage
-    !insertmacro skipPageIfUpdated
+    !ifmacrodef customLicensePagePre
+      !insertmacro customLicensePagePre
+    !else
+      !insertmacro skipPageIfUpdated
+    !endif
     !insertmacro licensePage
   !endif
diff --git a/templates/nsis/include/allowOnlyOneInstallerInstance.nsh b/templates/nsis/include/allowOnlyOneInstallerInstance.nsh
index b0937c5..2c14203 100644
--- a/templates/nsis/include/allowOnlyOneInstallerInstance.nsh
+++ b/templates/nsis/include/allowOnlyOneInstallerInstance.nsh
@@ -119,5 +119,2 @@
       ${endIf}
-      MessageBox MB_OKCANCEL|MB_ICONEXCLAMATION "$(appRunning)" /SD IDOK IDOK doStopProcess
-      Quit
-
       doStopProcess:
@@ -122 +119 @@
-      DetailPrint "$(appClosing)"
+      DetailPrint "$(closingApplication)"
diff --git a/templates/nsis/multiUser.nsh b/templates/nsis/multiUser.nsh
index 5b95e10..ddc04eb 100644
--- a/templates/nsis/multiUser.nsh
+++ b/templates/nsis/multiUser.nsh
@@ -84,3 +84,9 @@
       !endif
-
+      !ifdef APP_ARM64
+        ${if} ${RunningX64}
+        ${orif} ${IsNativeARM64}
+          StrCpy $0 "$PROGRAMFILES64"
+        ${endif}
+      !endif
+
       !ifdef MENU_FILENAME
